ocamldoc/Makefile

```diff
@@ -13,194 +13,240 @@
 #*                                                                        *
 #**************************************************************************
 
-include ../config/Makefile
-CAMLRUN ?= ../boot/ocamlrun
-CAMLYACC ?= ../boot/ocamlyacc
-
-# Various commands and dir
-##########################
-ROOTDIR   = ..
-OCAMLC    = $(CAMLRUN) $(ROOTDIR)/ocamlc -nostdlib -I $(ROOTDIR)/stdlib
-OCAMLOPT  = $(CAMLRUN) $(ROOTDIR)/ocamlopt -nostdlib -I $(ROOTDIR)/stdlib
-OCAMLDEP  = $(CAMLRUN) $(ROOTDIR)/tools/ocamldep
-OCAMLLEX  = $(CAMLRUN) $(ROOTDIR)/boot/ocamllex
-OCAMLLIB  = $(LIBDIR)
-OCAMLBIN  = $(BINDIR)
-
+ROOTDIR = ..
+
+include $(ROOTDIR)/config/Makefile
+OCAMLRUN ?= $(ROOTDIR)/boot/ocamlrun
+OCAMLYACC ?= $(ROOTDIR)/boot/ocamlyacc
+
+STDLIBFLAGS = -nostdlib -I $(ROOTDIR)/stdlib
+OCAMLC    = $(OCAMLRUN) $(ROOTDIR)/ocamlc $(STDLIBFLAGS)
+ifeq "$(UNIX_OR_WIN32)" "unix"
+OCAMLOPT  = $(OCAMLRUN) $(ROOTDIR)/ocamlopt $(STDLIBFLAGS)
+else # Windows
+  ifeq "$(wildcard $(ROOTDIR)/flexdll/Makefile)" ""
+    FLEXLINK_ENV=
+  else
+    FLEXLINK_ENV=OCAML_FLEXLINK="$(ROOTDIR)/boot/ocamlrun $(ROOTDIR)/flexdll/flexlink.exe"
+  endif
+  OCAMLOPT = $(FLEXLINK_ENV) $(OCAMLRUN) $(ROOTDIR)/ocamlopt $(STDLIBFLAGS)
+endif
+OCAMLDEP  = $(OCAMLRUN) $(ROOTDIR)/tools/ocamldep
+ifeq "$(UNIX_OR_WIN32)" "unix"
+OCAMLDEPFLAGS =
+else # Windows
+OCAMLDEPFLAGS = -slash
+endif
+OCAMLLEX  = $(OCAMLRUN) $(ROOTDIR)/boot/ocamllex
+# TODO: clarify whether the preprocessor really needs to be different here
```

Comment:
Alain Frisch (2016/09/13 14:36 -0700):
> You cannot simply do `-pp './remove_DEBUG'` under Windows because the
> command is executed with `cmd.exe` which does not know what to do with
> the shell script (but it knows how to call Cygwin's `grep` which is in
> the path).  The following should work, though: `-pp 'sh
> ./remove_DEBUG'`.

Indeed, thanks a lot for the suggestion. For sake of completeness,
it is worth mentionning that remove_DEBUG does not invoke grep but sed,
in order to preserve line numbers while removing the DEBUG lines.

The PR has been updated according to your suggestion, so that the
Windows build now uses the same preprocessor as the Unix one, since such
a unification really belongs to the idea of this PR.

> That said, I'm wondering whether one should just remove all those
> DEBUG lines (and get rid of the preprocessing).  Are they actually
> useful?

I don't know. I added a note about this in the Makefile because I think,
no matter what is decided, it should be implemented in a separate PR.

One alternative to mere removal could be to make it possible to enable
debugging at runtime rather than at compile time, e.g. by adding a
-debug option to ocamldoc (it does not have one yet).